### PR TITLE
Create masked input field

### PR DESCRIPTION
**Pull request by IgorZiegler**
- Add new text block which extends input block with default text type.
 - Also this text field has two specific mask parameters:
     1)  mask which represent a string with some mask template which use
         a - Represents an alpha character (A-Z,a-z)
         9 - Represents a numeric character (0-9)
         * - Represents an alphanumeric character (A-Z,a-z,0-9)
     2) maskPlacholder

### CODE REVIEW

File: public/examples/text/text.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      value: '1234 56789123',
+      rightMark: '<span class="glyphicon glyphicon-ruble" aria-hidden="true" style="font-size: 11px"></span>'
+    }]
+  });
+
+  console.log(form.items[0].value);
+  form.items[0].value = '100';
+  console.log(form.items[0].value);
+  form.items[0].value = '1123 1546789';
+  console.log(form.items[0].value);
```
Comment (amenshov) [inline public/examples/text/text.js]: it should be removed

File: frontend/blocks/text/text.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    this.textInput = this.el.find('input:text');
+    this.textInput.mask(this.config.mask, {placeholder: this.config.maskPlaceholder});
+  }
+
+  get value() {
+    const value = {};
+
+    value[this.config.name] = this.textInput.val();
+
+    return value;
```
Comment (amenshov) [inline frontend/blocks/text/text.js]:
```
return {
  [this.config.name]: this.textInput.val();
};
```

File: frontend/blocks/text/text.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import Input from '../input/input';
+import Factory from '../factory';
+import 'jquery.maskedinput/src/jquery.maskedinput';
+
+export default class Text extends Input {
+
+  render() {
+    super.render();
+
+    this.textInput = this.el.find('input:text');
```
Comment (amenshov) [inline frontend/blocks/text/text.js]: rename to `_input`

File: frontend/blocks/text/text.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  set value(val) {
+    this.textInput.val(val);
+  }
+
+  getTemplateData() {
+    const tplData = this.config;
+
+    tplData.type = 'text';
+  }
```
Comment (amenshov) [inline frontend/blocks/text/text.js]:
```
return Object.assign(super.getTemplateData(), {type: 'text'});
```

File: frontend/blocks/text/text.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import Input from '../input/input';
+import Factory from '../factory';
+import 'jquery.maskedinput/src/jquery.maskedinput';
```
Comment (amenshov) [inline frontend/blocks/text/text.js]: Are you sure that `import 'jquery.maskedinput'` doesn't work?

File: frontend/blocks/text/text.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    value[this.config.name] = this.textInput.val();
+
+    return value;
+  }
+
+  set value(val) {
+    this.textInput.val(val);
+  }
+
+  getTemplateData() {
```
Comment (maxmalov) [inline frontend/blocks/text/text.js]: Super, I like it :+1:

File: public/examples/text/text.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+// Some code to test out lib
+$(function () {
```
Comment (maxmalov) [inline public/examples/text/text.js]: You've included this script in the end of body

```js
(function () {
  // your boilerplate stuff here
})();
```

File: frontend/blocks/text/text.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import Input from '../input/input';
+import Factory from '../factory';
+import 'jquery.maskedinput/src/jquery.maskedinput';
+
+export default class Text extends Input {
+
+  render() {
+    super.render();
```
Comment (maxmalov) [inline frontend/blocks/text/text.js]:
```js
super.render(...arguments);
```

File: frontend/blocks/text/text.js (hunk 6 above)
Comment (maxmalov) [inline frontend/blocks/text/text.js]: Super, I like it :+1:

File: frontend/blocks/text/text.js (hunk 5 above)
Comment (IgorZiegler) [inline frontend/blocks/text/text.js]: Yes, I'm sure. I have checked it.

File: frontend/blocks/text/text.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import Input from '../input/input';
+import Factory from '../factory';
+import 'jquery.maskedinput/src/jquery.maskedinput';
+
+export default class Text extends Input {
+
+  render() {
+    super.render(...arguments);
```
Comment (amenshov) [inline frontend/blocks/text/text.js]: method `render` doesn't accept any parameters. Why do you pass them?

**Pull request merged**